File: typescript/src/problems201_300/contains-duplicate.ts
class Problem {
  containsDuplicate(nums: number[]): boolean {
    throw new Error(`Not implemented (${nums}).`);
  }
}

class Default extends Problem {
  containsDuplicate(nums: number[]): boolean {
    nums.sort((a, b) => a - b);
    for (let i = 1; i < nums.length; i++) {
      if (nums[i] === nums[i - 1]) {
        return true;
      }
    }
    return false;
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;

File: python/src/problems201_300/contains_duplicate.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def contains_duplicate(self, nums: list[int]) -> bool:
        pass


class Default(Problem):
    @override
    def contains_duplicate(self, nums: list[int]) -> bool:
        nums.sort()
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                return True
        return False


SOLUTIONS: list[Problem] = [Default()]
